File: SDKs/wealthos.cloud/v1/typescript/src/sdk/models/operations/getpotvalue.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetPotValuePathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=pot_id" })
  potId: string;
}


export class GetPotValueHeaders extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=x-api-key" })
  xApiKey: string;
}


export class GetPotValueSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=apiKey;subtype=header" })
  apiSecretKey: shared.SchemeApiSecretKey;
}


export class GetPotValueRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetPotValuePathParams;

  @SpeakeasyMetadata()
  headers: GetPotValueHeaders;

  @SpeakeasyMetadata()
  security: GetPotValueSecurity;
}


export class GetPotValueResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  getPotValue200ApplicationJSONAny?: any;

  @SpeakeasyMetadata()
  getPotValue400ApplicationJSONAny?: any;

  @SpeakeasyMetadata()
  getPotValue401ApplicationJSONAny?: any;

  @SpeakeasyMetadata()
  getPotValue403ApplicationJSONAny?: any;

  @SpeakeasyMetadata()
  getPotValue404ApplicationJSONAny?: any;

  @SpeakeasyMetadata()
  getPotValue429ApplicationJSONAny?: any;

  @SpeakeasyMetadata()
  getPotValue500ApplicationJSONAny?: any;
}

File: SDKs/wealthos.cloud/v1/python/src/sdk/models/operations/getpotvalue.py
import dataclasses
from typing import Any,Optional
from ..shared import security as shared_security


@dataclasses.dataclass
class GetPotValuePathParams:
    pot_id: str = dataclasses.field(metadata={'path_param': { 'field_name': 'pot_id', 'style': 'simple', 'explode': False }})
    

@dataclasses.dataclass
class GetPotValueHeaders:
    x_api_key: str = dataclasses.field(metadata={'header': { 'field_name': 'x-api-key', 'style': 'simple', 'explode': False }})
    

@dataclasses.dataclass
class GetPotValueSecurity:
    api_secret_key: shared_security.SchemeAPISecretKey = dataclasses.field(metadata={'security': { 'scheme': True, 'type': 'apiKey', 'sub_type': 'header' }})
    

@dataclasses.dataclass
class GetPotValueRequest:
    headers: GetPotValueHeaders = dataclasses.field()
    path_params: GetPotValuePathParams = dataclasses.field()
    security: GetPotValueSecurity = dataclasses.field()
    

@dataclasses.dataclass
class GetPotValueResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    get_pot_value_200_application_json_any: Optional[Any] = dataclasses.field(default=None)
    get_pot_value_400_application_json_any: Optional[Any] = dataclasses.field(default=None)
    get_pot_value_401_application_json_any: Optional[Any] = dataclasses.field(default=None)
    get_pot_value_403_application_json_any: Optional[Any] = dataclasses.field(default=None)
    get_pot_value_404_application_json_any: Optional[Any] = dataclasses.field(default=None)
    get_pot_value_429_application_json_any: Optional[Any] = dataclasses.field(default=None)
    get_pot_value_500_application_json_any: Optional[Any] = dataclasses.field(default=None)
    
